pkg/azure/plugin.go

```diff
@@ -847,6 +847,45 @@ func (s *azurePluginServer) AttachResource(ctx context.Context, attachResourceRe
 	return &paragliderpb.AttachResourceResponse{Name: *resource.Name, Uri: *resource.ID, Ip: networkInfo.Address}, nil
 }
 
+func (s *azurePluginServer) CheckOrFixResource(ctx context.Context, checkReq *paragliderpb.CheckResourceRequest, shouldFix bool) (*paragliderpb.CheckResourceResponse, error) {
+	checkResponse := &paragliderpb.CheckResourceResponse{Errors: []paragliderpb.ErrorCode{}}
+	resourceIdInfo, err := getResourceIDInfo(checkReq.Resource)
+	if err != nil {
+		return checkResponse, err
+	}
+	azureHandler, err := s.setupAzureHandler(resourceIdInfo, namespace)
+	if err != nil {
+		return checkResponse, err
+	}
+
+	// Check if the resource exists to validate the tags
+	resource, err := ValidateResourceExists(ctx, azureHandler, checkReq.Resource)
+	if err != nil {
+		// todo: Do this check in a different way
```

Comment:
I think there are other places in the code that do similar checks. Just make sure this is consistent with those for now.

Reply:
I'll look into this